smaily_for_drupal/src/Form/AdminForm.php

```diff
@@ -154,58 +155,73 @@ public function validateCredentials(array &$form, FormStateInterface $form_state
-            '<div class="messages messages--error">' . 'Credentials invalid.' . '</div>')
-        );
-        return $ajax_response;
-
-      case 404:
-        $ajax_response->addCommand(
-          new HtmlCommand(
-            '.smaily_message',
-            '<div class="messages messages--error">' . 'Subdomain error' . '</div>')
-        );
-        return $ajax_response;
-
-      default:
-        $ajax_response->addCommand(
-          new HtmlCommand(
-            '.smaily_message',
-            '<div class="messages messages--error">' . 'Something went wrong.' . '</div>')
-        );
-        return $ajax_response;
+    $url = 'https://' . $subdomain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted';
+    
+    try {
+      $client = \Drupal::httpClient();
+      $client->request('GET', $url, [
+        'auth' => [$username, $password],
+      ]);
+
+      $this->config('smaily_for_drupal.settings')
+        ->set('smaily_api_credentials.domain', $subdomain)
+        ->set('smaily_api_credentials.username', $username)
+        ->set('smaily_api_credentials.password', $password)
+        ->save();
+      $ajax_response->addCommand(
+        new HtmlCommand(
+          '.smaily_message',
+          '<div class="messages messages--status">' . 'Credentials valid' . '</div>')
+      );
+      return $ajax_response;
     }
+    catch (ClientException $e) {
+      switch ($e->getResponse()->getStatusCode()) {
+        case 200:
+          // OK code, continue validateCredentials.
+          break;
+
+        case 401:
+          $ajax_response->addCommand(
+            new HtmlCommand(
+              '.smaily_message',
+              '<div class="messages messages--error">' . 'Credentials invalid.' . '</div>')
+          );
+          return $ajax_response;
+
+        case 404:
+          $ajax_response->addCommand(
+            new HtmlCommand(
+              '.smaily_message',
+              '<div class="messages messages--error">' . 'Subdomain error' . '</div>')
+          );
+          return $ajax_response;
+
+        default:
+          $ajax_response->addCommand(
+            new HtmlCommand(
+              '.smaily_message',
+              '<div class="messages messages--error">' . 'Something went wrong.' . '</div>')
+          );
+          return $ajax_response;
+      }
 
-    $this->config('smaily_for_drupal.settings')
-      ->set('smaily_api_credentials.domain', $subdomain)
-      ->set('smaily_api_credentials.username', $username)
-      ->set('smaily_api_credentials.password', $password)
-      ->save();
-    $ajax_response->addCommand(
-      new HtmlCommand(
-        '.smaily_message',
-        '<div class="messages messages--status">' . 'Credentials valid' . '</div>')
-    );
-    return $ajax_response;
+      $credentials = [
+        '@username' => $username,
+        '@password' => $password,
+        '@subdomain' => $subdomain,
+      ];
+      \Drupal::logger('smaily_for_drupal')->error(
```

Comment:
Logging passwords is a bad practice.

Reply:
I didn't know what to log so I was more verbose. Should I keep the username/domain or drop it all and have a translateable string?

Reply:
Everything else is fine (subdomain and username), but you could also log the HTTP status code. Logging response body could be potentially bad if HTML content is returned.